src/backend/input.rs

```diff
@@ -0,0 +1,31 @@
+use std::hash::Hash;
+
+#[derive(Hash, PartialEq, Eq, Clone, Copy)]
+pub enum InputDevice<T: PartialEq + Eq + Hash + Clone + Copy>
```

Comment:
I think there would simply be one xkbcommon context per seat. And the idea is that `seat` is a logical representation, and libinput has a system-wide configuration setup apparently.

So basically, it would be up to the computer user to decide whether their two keyboard should be treated as a single one, or appear as two different logical seats.